shapeless/src/test/scala/zio/config/shapeless/AutomaticConfigDescriptorSpec.scala

```diff
@@ -0,0 +1,154 @@
+  private val genCredentials = Gen.oneOf(genToken, genPassword)
+
+  private val genDbUrl = genNonEmptyString(5).map(DbUrl)
+
+  private val genAws =
+    for {
+      region      <- genNonEmptyString(5)
+      credentials <- genCredentials
+    } yield Aws(region, credentials)
+
+  private[shapeless] val genEnvironment =
+    for {
+      aws            <- genAws
+      price          <- genPrice
+      dbUrl          <- genDbUrl
+      port           <- Gen.anyInt
+      amount         <- Gen.option(Gen.long(1, 100))
+      quantity       <- Gen.either(Gen.long(5, 10), genAlpha)
+      default        <- Gen.option(Gen.anyInt)
+      anotherDefault <- Gen.option(Gen.boolean)
+      descriptions   <- Gen.listOfBounded(1, 10)(Gen.anyString)
+      created        <- genLocalDateString
+      updated        <- genLocalTimeString
+      lastVisited    <- genLocalDateTimeString
+      id             <- Gen.anyUUID
+      partialMyConfig = Map(
+        "aws.region" -> aws.region,
+        aws.security match {
+          case Password(password) => "aws.security.credentials.password.value" -> password
+          case Token(token)       => "aws.security.credentials.token.value"    -> token
+        },
+        price match {
+          case Description(description) => "cost.price.description.value" -> description
+          case Currency(dollars)        => "cost.price.currency.value"    -> dollars.toString
+        },
+        "dburl.dburl"  -> dbUrl.dburl,
+        "port"         -> port.toString,
+        "quantity"     -> quantity.fold(d => d.toString, s => s),
+        "descriptions" -> descriptions.toString,
+        "created"      -> created,
+        "updated"      -> updated,
+        "lastVisited"  -> lastVisited,
+        "id"           -> id.toString
+      ) ++ amount.map(double => ("amount", double.toString)).toList
+    } yield (default, anotherDefault) match {
+      case (Some(v1), Some(v2)) => partialMyConfig ++ List(("default", v1.toString), ("anotherDefault", v2.toString))
+      case (Some(v1), None)     => partialMyConfig + (("default", v1.toString))
+      case (None, Some(v2))     => partialMyConfig + (("anotherDefault", v2.toString))
+      case (None, None)         => partialMyConfig
+    }
+
+  def genAlpha: Gen[Random, String] =
+    for {
+      n <- Gen.int(1, 10) // zio-config supports only cons hence starting with 1
+      s <- Gen.listOfN(n)(Gen.char(65, 122))
+    } yield s.mkString
+
+  val genInstant: Gen[Random, Instant] =
+    Gen.anyLong.map(Instant.ofEpochMilli)
+
+  val genLocalDateString: Gen[Random with Sized, String] =
+    genInstant.map(_.atZone(ZoneOffset.UTC).toLocalDate.toString)
+
+  val genLocalDateTimeString: Gen[Random with Sized, String] =
+    genInstant.map(_.atZone(ZoneOffset.UTC).toLocalDateTime.toString)
+
+  val genLocalTimeString: Gen[Random with Sized, String] =
+    genInstant.map(_.atZone(ZoneOffset.UTC).toLocalTime.toString)
+}
```

Comment:
👍

Reply:
All the tests are just copied from `magnolia`